test/regExpTest.coffee

```diff
@@ -0,0 +1,71 @@
+expect = require('chai').expect
+
+describe 'regExp matches URI', ->
+  regExp = new RegExp("^(http[s]?:\\/\\/(www\\.)?|ftp:\\/\\/(www\\.)?|www\\.){1}([0-9A-Za-z-\\.@:%_\+~#=]+)+((\\.[a-zA-Z]{2,3})+)(/(.)*)?(\\?(.)*)?", "i")
```

Comment:
This regex is incredibly complex, and doesn't even use things like `\d` and `\w` to make it easier to read. Why not simply:
```
[a-z]+[:.].*?(?=\s)
```